### ISSUE TITLE
conda-forge recipe feedback

### ISSUE THREAD

**Issue by luizirber** [stjude-rust-labs/sprocket#42]
Hi!

I started a conda-forge recipe for sprocket in
https://github.com/conda-forge/staged-recipes/pull/28240

It is already working for Linux and macOS (x86_64), will fix windows build for this PR and then work on Linux/macOS aarch64 once the staged recipe is merged and a feedstock is available.

I made it for `0.8.0` because `0.9.0` needs Rust `1.82`, but conda-forge [only has 1.81 at the moment](https://github.com/conda-forge/rust-feedstock/pull/216).

I think it would be good to have a dev from your team as recipe maintainer too, please let me know if you have volunteers =]

**Comment by a-frantz** [comment]
We appreciate the interest in contributing to the project!

However we aren't convinced it's wise to take on the technical debt of maintaining a conda-forge recipe. I personally am the maintainer of a handful of Python packages and it can spiral into a fair amount of work to get the build right. I do not have experience with any Rust binary recipes, but the fact you are running into MSRV issues doesn't inspire confidence.

We think the current install experience is straightforward (`cargo install sprocket` or downloading the published binaries on the release page) and have plans to add a Homebrew tap once we reach the GitHub Star threshold. It's unclear what being on conda-forge offers in terms of easing the install experience.

We are also planning on rolling our own Docker container image and publishing to ghcr.io (#43) so the appeal of having conda-forge handle that for us isn't there.

Are there any specific reasons you want sprocket to be on conda-forge that I might have missed?

**Comment by luizirber** [comment]
> However we aren't convinced it's wise to take on the technical debt of maintaining a conda-forge recipe. I personally am the maintainer of a handful of Python packages and it can spiral into a fair amount of work to get the build right. I do not have experience with any Rust binary recipes, but the fact you are running into MSRV issues doesn't inspire confidence.

1.82 was published this week. Distributions and packaging might lag a bit on the newest Rust compiler, so I'm not surprised 1.82 was not there yet.

The conda recipe turned out to be very easy to write, thanks for removing the openssl dep on 0.9.0!

> We think the current install experience is straightforward (`cargo install sprocket` or downloading the published binaries on the release page) and have plans to add a Homebrew tap once we reach the GitHub Star threshold. It's unclear what being on conda-forge offers in terms of easing the install experience.

`cargo install` requires a rust compiler, downloading published binaries require customized setup. Homebrew doesn't work on windows or linux arm64 (yet?), and involves more system changes to install and use.

> Are there any specific reasons you want sprocket to be on conda-forge that I might have missed?

While there is no perfect solution, I've been really enjoying [pixi](https://pixi.sh) for maintaining project-level dependencies, which allows running the same versions across local dev, CI, and docker images. And the pre-req for using pixi is having a conda package available, so that's why I created one.

I did it in conda-forge because it has better support for multiple platforms, bioconda just recently got support for linux-aarch64 and osx-arm64 (and no win-64 in sight). It is also pretty easy to manage a feedstock in conda-forge once it is created.

---

I asked if someone wanted to be in the maintainer list for the feedstock because I like to track packaging for my software and help if needed, but you don't need to do anything if you don't think this is not advantageous to you. I'll watch for new releases here and update as needed.

The conda-forge package now has 0.9.0, and builds for linux (aarch4 and x86_64), macos (arm64 and x86_64), and windows (x86_64), so I'm closing this issue.
